### PR TITLE
test: increase test coverage for email validation

### PR DESCRIPTION
**Pull request by favna**
TODO:
- [x] The statement `if (email.length - lastDotIndex > 63) return false;` is currently uncovered and I also have no idea what input to give to cover it. Please advice.
- [x] Giving `short-account-name@domain-name.-that-has-more-than-sixty-three-characters.com` as input throws a `ReferenceError`, unknown why. This should be solved.

### CODE REVIEW

File: tests/jest.setup.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+function fail(reason = 'Expected to throw, but failed to do so') {
+	throw new Error(reason);
+}
+
+global.fail = fail;
```
Comment (kyranet) [inline tests/jest.setup.ts:5]:
```suggestion
global.fail ??= fail;
```

File: tests/jest.setup.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+function fail(reason = 'Expected to throw, but failed to do so') {
```
Comment (kyranet) [inline tests/jest.setup.ts:1]:
```suggestion
function fail(reason = 'fail was called in a test.') {
```

File: tests/common/macros/comparators.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export function expectModifiedClonedValidator<T>(expected: BaseValidator<T>, act
 export function expectError<T = any>(cb: () => T, expected: BaseError) {
 	try {
 		cb();
-		fail('expected to throw, but failed to do so');
+		fail();
```
Comment (kyranet) [inline tests/common/macros/comparators.ts:32]: This should be moved outside the try/catch block, and be called with `'Expected to throw, but failed to do so'`

Comment (kyranet) [reply tests/common/macros/comparators.ts:32]: A `return;` at the end of the catch block allows us to not call `fail()` if the callback threw.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: tests/validators/string.test.ts [hunk 4]
```diff
@@ -112,12 +112,29 @@ describe('StringValidator', () => {
...
+				'short-account-name@domain-name-that-has-more-than-sixty-three-characters-and-is-then.followed-by-another-segment-that-got-split-by-a-full-stop-symbol.com',
+				`foo@bar.${'a'.repeat(64)}`
+			])('GIVEN %p THEN throws a ConstraintError', (input) => {
 				expectError(
 					() => emailPredicate.parse(input),
 					new ExpectedConstraintError('s.string.email', 'Invalid email address', input, 'expected to be an email address')
 				);
 			});
+
+			test.each([undefined, null])('GIVEN %p THEN throws a ValidationError', (input) => {
```
Comment (kyranet) [inline tests/validators/string.test.ts:135]: This doesn't touch the constraint, it needs to be in the validator's parse tests, at line 11.

**Review by kyranet — CHANGES_REQUESTED** [review]
Just this one and LGTM.

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**